File: programmers/lev1/77484.py
def solution(lottos, win_nums):
    answer = []
    lotto_dic = dict()

    win_cnt = 0

    cnt_0 = 0

    for lotto in lottos:
        if lotto in win_nums :
            win_cnt += 1
        if lotto == 0:
            cnt_0 += 1
    # print(win_cnt, cnt_0)

    if (win_cnt + cnt_0) < 2:
        answer.append(6)
    else:
        answer.append(7 - win_cnt - cnt_0)
    
    if 7 - win_cnt >= 6:
        answer.append(6)
    else:
        answer.append(7 - win_cnt)
    
    # print(answer)
    return answer

lottos = [44, 1, 0, 0, 31, 25]
win_nums = [31, 10, 45, 1, 6, 19]
# lottos = [1,2,3,4,5,6]
# win_nums = [7,8,9,10,11,12]
solution(lottos, win_nums)

File: java/src/programmers/year2023/lv1/_77484.java
package programmers.year2023.lv1;

import java.util.HashMap;

public class _77484 {
    public static void main(String[] args) {
//        int[] lottos = {44,1,0,0,31,25};
//        int[] win_nums = {31, 10, 45, 1, 6, 19};

//        int[] lottos = {0,0,0,0,0,0};
//        int[] win_nums = {38,19,20,40,15,25};

//        int[] lottos = {45,4,35,20,3,9};
//        int[] win_nums = {20,9,3,45,4,35};

        int[] lottos = {21,10,7,46,5,36};
        int[] win_nums = {20,9,3,45,4,35};


        solution(lottos, win_nums);
    }

    public static int[] solution(int[] lottos, int[] win_nums) {
        int correctCnt = 0;
        int zeroCnt = 0;
        HashMap<Integer, Boolean> win = new HashMap<>();
        for (int lotto : lottos) {
            if (lotto != 0) {
                win.put(lotto, true);
            }
            if (lotto == 0) {
                zeroCnt += 1;
            }
        }
        for (int win_num : win_nums) {
            if (win.containsKey(win_num)) {
                correctCnt += 1;
            }
        }
        System.out.println("correctCnt = " + correctCnt);
        System.out.println("zeroCnt = " + zeroCnt);
        System.out.println("top:" +(7 - correctCnt - zeroCnt));
        System.out.println(":" + (correctCnt >= 2 ? 7 - correctCnt : 6));

        return new int[]{Math.min(7 - correctCnt - zeroCnt, 6), correctCnt >= 2 ? 7 - correctCnt : 6};
    }
}
